src/tests/test_max_lhs_option.cpp

```diff
@@ -0,0 +1,57 @@
+#include <filesystem>
+
+#include <gtest/gtest.h>
+
+#include "algorithms/algo_factory.h"
+#include "algorithms/fd/fastfds/fastfds.h"
+#include "algorithms/fd/fd_algorithm.h"
+#include "algorithms/fd/tane/tane.h"
+#include "config/names.h"
+#include "datasets.h"
+#include "model/table/relational_schema.h"
+
+using algos::FastFDs, algos::Tane;
+using config::InputTable;
+
+InputTable GetInputTable(const std::filesystem::path& path, char separator = ',',
+                         bool has_header = true) {
+    return std::make_unique<CSVParser>(path, separator, has_header);
+}
+
+template <typename T>
+class MaxLHSOptionTest : public ::testing::Test {
+protected:
+    static std::unique_ptr<algos::FDAlgorithm> CreateAlgoInstance(unsigned int max_lhs,
+                                                                  bool EqNullsType,
+                                                                  const std::filesystem::path& path,
+                                                                  char separator = ',',
+                                                                  bool hasHeader = true) {
+        using namespace config::names;
+
+        algos::StdParamsMap params{
+                {kCsvPath, path},        {kSeparator, separator},
+                {kHasHeader, hasHeader}, {kTable, GetInputTable(path, separator, hasHeader)},
+                {kMaximumLhs, max_lhs},  {kEqualNulls, EqNullsType}};
+
+        return algos::CreateAndLoadAlgorithm<T>(params);
+    }
+};
+
+TYPED_TEST_SUITE_P(MaxLHSOptionTest);
+
+TYPED_TEST_P(MaxLHSOptionTest, MaxLHSOptionWork) {
+    unsigned int max_lhs = 1;
+    auto const path = test_data_dir / "TestFD.csv";
+    auto algorithm = TestFixture::CreateAlgoInstance(max_lhs, true, path);
+    algorithm->Execute();
+    std::list<FD> result_fds_list = algorithm->FdList();
+    for (auto& fd : result_fds_list) {
+        ASSERT_TRUE(fd.GetLhs().GetArity() <= max_lhs);
+    }
+}
+
+REGISTER_TYPED_TEST_SUITE_P(MaxLHSOptionTest, MaxLHSOptionWork);
+
+using Algorithms = ::testing::Types<algos::Tane, algos::FastFDs>;
```

Comment:
This test passes only because `Tane` and `FastFDs` treat `max_lhs` correctly internally during mining process, try `HyFD` which doesn't do anything like this and the test will fail. The goal is to make all algorithms working with `max_lhs` option even when they know nothing about it.
So this test should pass on _all_ fd mining algorithms. Also, there is no need to create a separate file with the test for this single option, add a test case to `src/tests/test_algorithm.cpp` instead (there all tests for fd mining algorithms are implemented)